django/db/models/expressions.py

```diff
@@ -1251,6 +1254,28 @@ def __repr__(self):
         return "{}({})".format(self.__class__.__name__, self.expression)
 
 
+class NegatedExpression(ExpressionWrapper):
+    """
+    Wrapping an expression to negate its output.
+    """
+
+    def __init__(self, expression):
+        super().__init__(expression, output_field=fields.BooleanField())
+
+    def __invert__(self):
+        return self.expression
```

Comment:
_Gut feeling_ says that this ought to be a _clone_, so as not to leak the _original_ (un-negated) mutable expression out.

e.g something like:
```
return self.expression.copy()
```

Currently you've got a test for `self.assertEqual(~~c, c)` which I would _guess_ (reading the diff only) would also pass with `assertIs` where I presume it oughtn't.